File: src/main/java/com/lannstark/lec07/Lec07Main.java
package com.lannstark.lec07;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public class Lec07Main {

  public static void main(String[] args) throws IOException {
    Lec07Main lec07Main = new Lec07Main();
    lec07Main.parseIntOrThrow("10");
    lec07Main.parseIntOrThrow("AA");
  }

  private int parseIntOrThrow(@NotNull String str) {
    try {
      return Integer.parseInt(str);
    } catch (NumberFormatException e) {
      throw new IllegalArgumentException(String.format("주어진 %s는 숫자가 아닙니다.", str));
    }
  }

}

File: src/main/java/com/lannstark/lec07/Lec07Main.kt
package com.lannstark.lec07

import java.io.BufferedReader
import java.io.FileReader

fun main() {
    parseIntOrThrow("10");
    // parseIntOrThrow("AA");

    readFile("./a.txt")

}

fun parseIntOrThrow(str: String): Int {
    try {
        return str.toInt()
    }catch (e: NumberFormatException) {
        throw IllegalArgumentException("주어진 ${str}는 숫자가 아닙니다.");
    }
}

fun readFile(path: String) {
    BufferedReader(FileReader(path)).use { reader ->
        println(reader.readLine())
    }
}